### ISSUE TITLE
dataframe with None in a bool column triggers an error

### ISSUE THREAD

**Issue by julioevi** [rpy2/rpy2#916]
**Describe the issue or bug**

Passing a dataframe with `None` in a bool column triggers an error

**To Reproduce**

```python
import pandas as pd
lst = [None, False, True]
df = pd.DataFrame(lst)
df
```

then on a new cell

```pyton
%%R -i df
```

**Expected behavior**
no error

**Error**
```python
KeyError                                  Traceback (most recent call last)
/opt/conda/lib/python3.8/site-packages/rpy2/robjects/pandas2ri.py in py2rpy_pandasdataframe(obj)
     56         try:
---> 57             od[name] = conversion.converter_ctx.get().py2rpy(values)
     58         except Exception as e:

/opt/conda/lib/python3.8/functools.py in wrapper(*args, **kw)
    874 
--> 875         return dispatch(args[0].__class__)(*args, **kw)
    876 

/opt/conda/lib/python3.8/site-packages/rpy2/robjects/pandas2ri.py in py2rpy_pandasseries(obj)
    201         # TODO: Could this be merged with obj.type.name == 'O' case above ?
--> 202         res = _PANDASTYPE2RPY2[homogeneous_type](obj)
    203     elif type(obj.dtype) in (pandas.Float64Dtype, pandas.BooleanDtype):

KeyError: <class 'bool'>

... more trace msgs ...

AttributeError: 'bool' object has no attribute 'encode'

```

**Additional context**
This bug is new in 3.5.3
Python 3.8.8
R version 4.0.5
Ubuntu 18.04.5 LTS

**Comment by lgautier** [comment]
It does fail to fall back to string conversion somehow.

Note that this might not be what you'd hope though. The `pandas.Series` in the data frame
is of `dtype` object, meaning any list of Python object:

```python
>>> df.dtypes 
0    object
dtype: object
```

Note the different `dtype` here:

```python
import pandas as pd
lst = [True, False, True]
df = pd.DataFrame(data=lst)
```

```python
>>> df.dtypes
0    bool
dtype: object
```

Specify the type in `pandas` works without errror:

```python
df = pd.DataFrame(data=lst, dtype=pd.BooleanDtype())
```

**Comment by lgautier** [comment]
It does fail to fall back to string conversion somehow.

Note that this might not be what you'd hope though. The `pandas.Series` in the data frame
is of `dtype` object, meaning any list of Python object:

```python
>>> df.dtypes 
0    object
dtype: object
```

Note the different `dtype` here:

```python
import pandas as pd
lst = [True, False, True]
df = pd.DataFrame(data=lst)
```

```python
>>> df.dtypes
0    bool
dtype: object
```

It works when specifying the type in `pandas` with:

```python
df = pd.DataFrame(data=lst, dtype=pd.BooleanDtype())
```

**Comment by julioevi** [comment]
thanks for the pointer, I'll give it a try

**Comment by lgautier** [comment]
Fixed with PR #951.
